maro/simulator/scenarios/finance/common/trader.py

```diff
@@ -0,0 +1,252 @@
+import math
+from enum import Enum
+from collections import OrderedDict
+from maro.simulator.scenarios.finance.common.common import (
+    Action, OrderMode
+)
+
+
+class TradeConstrain(Enum):
+    min_buy_unit = "min_buy_unit"
+    min_sell_unit = "min_sell_unit"
+
+
+class Trader():
+    def __init__(self, trade_constrain: dict):
+        self._order_handlers = OrderedDict()
+        self._slippage_handler = None
+        self._commission_handlers = []
+        self._trade_constrain = OrderedDict()
+        for constrain in trade_constrain:
+            self._trade_constrain[constrain] = trade_constrain[constrain]
+
+        self._trade_number = 0
+
+    @property
+    def supported_orders(self) -> list:
+        return self._order_handlers.keys()
+
+    def order_handler_register(self, order_type: OrderMode, order_handler: callable):
+        self._order_handlers[order_type] = order_handler
+
+    def slippage_handler_register(self, slippage_handler: callable):
+        self._slippage_handler = slippage_handler
+
+    def commission_handler_register(self, commission_handler: callable):
+        self._commission_handlers.append(commission_handler)
+
+    def trade(self, order_action: Action, cur_data: dict, remaining_money: float):
+        # return stock, success, stock_price, number, tax
+        asset = order_action.item_index
+        is_success = False
+        is_trigger = True
+
+        deal_volume = abs(order_action.number)
+        is_buy = True if order_action.number >= 0 else False
+
+        deal_price = cur_data[order_action.item_index].opening_price
+        if 'deal_price' in self._trade_constrain:
```

Comment:
rename constrain to constraint

Reply:
done

Reply:
I didn't find the trader in new repo, where can I find the logic of what the previous trader does?

Reply:
there are still typo for this word. You use ``constrant`` in the code. It should be ``constraint``.